pkg/load/load.go

```diff
@@ -32,6 +47,48 @@ func Config(path string) (*api.ReleaseBuildConfiguration, error) {
 	if err := yaml.Unmarshal([]byte(raw), configSpec); err != nil {
 		return nil, fmt.Errorf("invalid configuration: %v\nvalue:\n%s", err, raw)
 	}
+
+	// If configresolver address info is provided, GET a copy of the config from there
+	if info != nil {
+		req, err := http.NewRequest("GET", fmt.Sprintf("http://%s/config", info.Address), nil)
+		if err != nil {
+			logrus.WithError(err).Error("Failed to create request for configresolver")
+			return configSpec, nil
+		}
+		query := req.URL.Query()
+		query.Add("org", info.Org)
+		query.Add("repo", info.Repo)
+		query.Add("branch", info.Branch)
+		if len(info.Variant) > 0 {
+			query.Add("variant", info.Variant)
+		}
+		req.URL.RawQuery = query.Encode()
+		client := &http.Client{}
+		resp, err := client.Do(req)
+		if err != nil {
+			logrus.WithError(err).Errorf("Failed to make request to configresolver")
+			return configSpec, nil
+		}
+		if resp.StatusCode != http.StatusOK {
+			logrus.Errorf("Response from configresolver != %d", http.StatusOK)
+			return configSpec, nil
+		}
+		data, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			logrus.WithError(err).Error("Failed to read configresolver response body")
+			return configSpec, nil
+		}
+		configSpecHTTP := &api.ReleaseBuildConfiguration{}
+		err = json.Unmarshal(data, configSpecHTTP)
+		if err != nil {
+			logrus.Errorf("failed to unmarshal config from configresolver: invalid configuration: %v\nvalue:\n%s", err, string(data))
+			return configSpec, nil
+		}
+		if !reflect.DeepEqual(configSpec, configSpecHTTP) {
```

Comment:
Just a note: If we enable this in production, we'll see this failing for rehearsals running over a changed config (where a "production" resolver will have the master config, but the job itself will get the changed one).

Reply:
Yeah rehearsals would need to be still loading from disk from the mutated configmap we pass in, that's a known gap